src/parser/ast/debug_sym_factory.rs

```diff
@@ -1,13 +1,13 @@
-use std::rc::Rc;
+use std::sync::Arc;
 
-use codespan_reporting::files::{Files, SimpleFile};
+use codespan_reporting::files::SimpleFile;
 
 use super::DebugSymRef;
 
 /// Factory for creating debug symbol references.
 pub struct DebugSymRefFactory {
     /// Source file reference.
-    file: Rc<SimpleFile<String, String>>,
+    file: Arc<SimpleFile<String, String>>,
```

Comment:
@alxkzmn you introduced `Arc` in case we are doing multi-threading compilation?

Reply:
No, this is because the language server is inherently asynchronous, so if it shares the code with the comoiler it won't compile with Rc.

Reply:
Ok :+1: